visualize_golden_test.go

```diff
@@ -53,3 +53,24 @@ func VerifyVisualization(t *testing.T, testname string, c *Container) {
 	assert.Equal(t, want, got,
 		"Output did not match. Make sure you updated the testdata by running 'go test -generate'")
 }
+
+func VerifyVisualizationError(t *testing.T, testname string, e error) {
+	var b bytes.Buffer
+	require.NoError(t, VisualizeError(e, &b))
+
+	dotFile := filepath.Join("testdata", testname+".dot")
+
+	if *generate {
+		err := ioutil.WriteFile(dotFile, b.Bytes(), 0644)
+		require.NoError(t, err)
+		return
+	}
+
+	wantBytes, err := ioutil.ReadFile(dotFile)
+	require.NoError(t, err)
+
+	got := b.String()
+	want := string(wantBytes)
+	assert.Equal(t, want, got,
+		"Output did not match. Make sure you updated the testdata by running 'go test -generate'")
+}
```

Comment:
This looks like it creates a file with the bytes written into the `bytes.Buffer` and asserts that the result of reading that file are the same bytes. Will this ever fail?

Did you instead mean to compare this against some fixture?

Reply:
The bytes will only be written to the file if the `generate` flag is on when running `go test`, otherwise it will be compared against existing files in the `testdata/` directory.

Reply:
*facepalm* Ah yea, my mistake. Thanks for clarifying!